src/logger.service.ts

```diff
@@ -23,148 +17,141 @@ const Levels = [
-        } catch (e) {
-          return `The additional[${idx}] value could not be parsed using JSON.stringify().`;
-        }
-      });
-    }
-
-    this.http.post(this.options.serverLoggingUrl,
-      {
-        level: Levels[level],
-        message: messageToLog,
-        additional: additionalToLog,
-        timestamp: this._timestamp()
-      },
-      {
-        headers
-      })
-      .subscribe(
-        (res: any) => this._log(NgxLoggerLevel.TRACE, false, 'Server logging successful', [res]),
-        (error: HttpErrorResponse) => this._log(NgxLoggerLevel.ERROR, false, 'FAILED TO LOG ON SERVER', [error])
-      );
+  public updateConfig(config: LoggerConfig) {
+    this.configService.updateConfig(config);
   }
 
-  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = []): void {
+  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = [], timestamp: string): void {
     switch (level) {
       case NgxLoggerLevel.WARN:
-        console.warn(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.warn(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.ERROR:
-        console.error(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.error(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.INFO:
-        console.info(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.info(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       default:
-        console.log(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.log(`${timestamp} [${Levels[level]}] `, message, ...additional);
     }
   }
 
-  private _log(level: NgxLoggerLevel, logOnServer: boolean, message, additional: any[] = []): void {
-    if (!message) {
-      return;
-    }
-
+  private _prepareMessage(message) {
     try {
       if (message instanceof Error) {
         message = message.stack;
       } else if (typeof message !== 'string') {
         message = JSON.stringify(message, null, 2);
       }
     } catch (e) {
-      additional = [message, ...additional];
+      // additional = [message, ...additional];
       message = 'The provided "message" value could not be parsed with JSON.stringify().';
     }
 
-    // Allow logging on server even if client log level is off
-    if (logOnServer) {
-      this._logOnServer(level, message, additional);
+    return message;
+  }
+
+  private _prepareAdditionalParameters(additional: any[]) {
+    if (additional === null || additional === undefined) {
+      return [];
     }
 
+    const additionalParams: any[] = [];
+
+    for (let i = 0; i < additional.length; i++) {
+      try {
+        if (typeof additional[i] === 'object') {
+          additional[i] = JSON.stringify(additional[i], null, 2);
+        }
```

Comment:
What are you doing here if the value is an Array?

Reply:
typeof [] === 'object', so this conditional handles both objects and arrays. It will try to stringify them both.